tds_test.go

```diff
@@ -694,6 +694,59 @@ func TestBadCredentials(t *testing.T) {
 	_ = testConnectionBad(t, params.URL().String())
 }
 
+func TestLeakedConnections(t *testing.T) {
+	goodParams := testConnParams(t)
+	badParams := testConnParams(t)
+	badParams.Database = "unknown_db"
+
+	// Connecting with good credentials should not fail
+	goodConn, err := sql.Open("sqlserver", goodParams.URL().String())
+	if err != nil {
+		t.Fatal("Open connection failed:", err.Error())
+	}
+	err = goodConn.Ping()
+	if err != nil {
+		t.Fatal("Ping with good credentials should not fail, but got error:", err.Error())
+	}
+
+	var localNetAddr string
+	err = goodConn.QueryRow("SELECT local_net_address FROM sys.dm_exec_connections WHERE session_id=@@SPID").Scan(&localNetAddr)
```

Comment:
you need an `ISNULL` on local_net_address to return an empty string or use one of the nullable types for the scan.